src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,375 @@
+
+                        startIndex = startIndex + BATCH_SIZE;
+                    }
+                    while (endIndex != lines.size() - 1 && startIndex < lines.size());
+
+                    featureSqlStatement.close();
+                    flagSqlStatement.close();
+
+                }
+                catch (final IOException error)
+                {
+                    logger.error("Exception while reading {}:", file, error);
+                }
+                catch (final SQLException error)
+                {
+                    logger.error("Exception batch executing flag statements", error);
+                }
+            });
+        });
+
+        logger.info("Atlas Checks database upload command finished in {} seconds.",
+                timer.elapsedSince().asSeconds());
+
+        return 0;
+    }
+
+    @Override
+    public String getCommandName()
+    {
+        return "flag-database";
+    }
+
+    @Override
+    public String getSimpleDescription()
+    {
+        return "Upload Atlas Checks flags into a Postgres database";
+    }
+
+    @Override
+    public void registerManualPageSections()
+    {
+        this.addManualPageSection("DESCRIPTION", FlagDatabaseSubCommand.class
+                .getResourceAsStream("FlagDatabaseSubCommandDescriptionSection.txt"));
+        this.addManualPageSection("EXAMPLES", FlagDatabaseSubCommand.class
+                .getResourceAsStream("FlagDatabaseSubCommandExamplesSection.txt"));
+    }
+
+    @Override
+    public void registerOptionsAndArguments()
+    {
+        this.registerOptionWithRequiredArgument(FLAG_PATH_INPUT, 'f',
+                "A directory of folders containing atlas-checks log files.",
+                OptionOptionality.REQUIRED, FLAG_PATH_INPUT);
+        this.registerOptionWithRequiredArgument(DATABASE_URL_INPUT, 't',
+                "Database connection string", OptionOptionality.REQUIRED, DATABASE_URL_INPUT);
+        this.registerOptionWithRequiredArgument(COUNTRIES_INPUT, 'i',
+                "List of iso3 country codes to filter by", OptionOptionality.OPTIONAL,
+                COUNTRIES_INPUT);
+        this.registerOptionWithRequiredArgument(CHECKS_INPUT, 'c', "A folder to output results to.",
+                OptionOptionality.OPTIONAL, CHECKS_INPUT);
+        super.registerOptionsAndArguments();
+    }
+
+    /**
+     * @param sql
+     *            PreparedStatement to add parameterized values to
+     * @param flag
+     *            CheckFlag to associate with feature
+     * @param feature
+     *            - Flagged AtlasItem
+     */
+    private void batchFlagFeatureStatement(final PreparedStatement sql, final CheckFlag flag,
+            final JsonObject feature)
+    {
+
+        final JsonObject properties = feature.get(PROPERTIES).getAsJsonObject();
+
+        try
+        {
+            sql.setString(1, flag.getIdentifier());
+            sql.setString(2, feature.get("geometry").toString());
```

Comment:
mega-nit: Could we do something like `GEOMETRY_INDEX`, `OSM_ID_INDEX`,  `ID_INDEX`, etc instead of `2`, `THREE`, `FOUR`, etc?

Reply:
The indices don't really have any significance outside of "this is the order in which the insert statement string contains each column".
For example,
```sql
INSERT INTO flag(flag_id, check_name, instructions, date_created) VALUES (?,?,?,?)
```
Given that, I don't think it's necessary to give these values names. Especially since 3 & 4 are used twice, but for two completely different column names